### DISCUSSION TITLE
[V3] Repeatable rules in form objects

### DISCUSSION THREAD

**Discussion by RikvdHeijden** [livewire/livewire#6142] [7. Bugs (archived)]
### Livewire version

3

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a From Object with multiple Attribute Rules per element.
2. Try to validate it.
3. An exception occurs

I've written a unit test that recreates my issue

```php
<?php

namespace Livewire\Features\SupportValidation;

use Livewire\Form;
use Livewire\Livewire;
use Tests\TestComponent;

class TestForm extends Form
{
    #[Rule('required')]
    #[Rule('min:2')]
    #[Rule('max:3')]
    public $foo = '';

    #[
        Rule('sometimes'),
        Rule('max:1')
    ]
    public $bar = '';
}

class UnitTest extends \Tests\TestCase
{
    /** @test */
    public function base_rule_attributes_merge_error()
    {

        Livewire::test(new class extends TestComponent {
            public TestForm $form;
        })
            ->set('form.foo', '')
            ->assertHasErrors(['form.foo' => 'required'])
            ->set('form.foo', '1')
            ->assertHasErrors(['form.foo' => 'min'])
            ->set('form.foo', '12345')
            ->assertHasErrors(['form.foo' => 'max'])
            ->set('form.foo', 'ok')
            ->assertHasNoErrors()
            ->set('form.bar', '12')
            ->assertHasErrors(['form.bar' => 'max'])
            ->set('form.bar', '1')
            ->assertHasNoErrors()
        ;
    }
}
```

This test fails with the exception: "Argument #1 ($string) must be of type string, array given."

### Additional information

In my tests it seems that in this case the array_merge_recursive in the HandlesValidation::addValidationAttributesFromOutside causes this issue. Changing that naively to array_merge fixes this issue and seems to work until I try to add custom messages to my rules like below:

```php
<?php

namespace Livewire\Features\SupportValidation;

use Livewire\Form;
use Livewire\Livewire;
use Tests\TestComponent;

class TestForm extends Form
{
    #[Rule('required', message: 'test')]
    #[Rule('min:2', message: 'test2')]
    #[Rule('max:3', message: 'test3')]
    public $foo = '';

    #[
        Rule('sometimes'),
        Rule('max:1')
    ]
    public $bar = '';
}

class UnitTest extends \Tests\TestCase
{
    /** @test */
    public function base_rule_attributes_merge_error_messages()
    {
        $component = Livewire::test(new class extends TestComponent {
            public TestForm $form;
        });

        $messages = $component
            ->set('form.foo', '')
            ->assertHasErrors(['form.foo' => 'required'])
            ->errors()->getMessages();

        $this->assertContains('test', $messages);

        $messages = $component
            ->set('form.foo', '1')
            ->assertHasErrors(['form.foo' => 'min'])
            ->errors()->getMessages();

        $this->assertContains('test2', $messages);

        $messages = $component
            ->set('form.foo', '12345')
            ->assertHasErrors(['form.foo' => 'max'])
            ->errors()->getMessages();

        $this->assertContains('test3', $messages);

        $component
            ->set('form.foo', 'ok')
            ->assertHasNoErrors()
        ;
    }
}
```

Now I always get the same error message regardless of the validation that failed. 

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by muffycompo** [comment]
@RikvdHeijden I am having the same issue so I imagine it was the repeatable rule [addition](https://github.com/livewire/livewire/pull/5876) as from beta-5. My example below is exactly what you described too:

```php
<?php

declare(strict_types=1);

namespace App\Livewire\Forms\MyApp;

use Livewire\Form;
use Livewire\Attributes\Rule;

class MyAppForm extends Form
{
    #[Rule(
        rule: [
            'required'
        ],
        as: 'attribute type', // <-- adding this throws the exception mentioned and removing it reverts back to form.attribute_type 
        onUpdate: false
    )]
    public ?string $attribute_type = null;
}
```

After struggling to find ways around this exception, I reverted back to beta-4 (before the addition of repeatable rule [pull request](https://github.com/livewire/livewire/pull/5876)). @nuernbergerA have you come across the aforementioned issue?

Cheers!

**Reply by nuernbergerA** [reply]
Yeah you'd be due the naive implementation I did.

Can you PR a failing test, so we can fix it?

**Reply by muffycompo** [reply]
Sure @nuernbergerA, I have opened a [PR](https://github.com/livewire/livewire/pull/6217) with a failing test as requested.

Thanks again for looking into this.

Cheers!

**Comment by citricguy** [comment]
Related to [6187](https://github.com/livewire/livewire/discussions/6187)?

If there are multiple rules and they have custom messages exceptions are thrown.

Edit: Actually, my bug at 6187 looks like a duplication of this one.
